### PR TITLE
Update: replace legacy async code and BLE unpack code with the latest…

### PR DESCRIPTION
**Pull request by JeremyYu-cn**
Update: To replace legacy async code and BLE unpack code with the latest version.

### CODE REVIEW

**Comment by taulfsime** [comment]
LGTM, @Mircho it's your call

**Review by Mircho — COMMENTED** [review]
Thank you, good contribution.

Few minor comments.

File: ble-shelly-btn-gateway-for-other-devices.js [hunk 1]
```diff
@@ -128,50 +137,65 @@ let BTHomeDecoder = {
...
+            Shelly.call("HTTP.GET", {
+                url: url,
                 timeout: 5
-                }, 
-                function(_, error_code, _, data) {
-                    if(error_code !== 0) {
+            },
+                function (_, error_code, _, data) {
+                    if (error_code !== 0) {
                         console.log("Calling", data.url, "failed");
```
[the remark this answers is not in the document]
Comment (JeremyYu-cn) [reply ble-shelly-btn-gateway-for-other-devices.js:180]: Changed.

**Comment by JeremyYu-cn** [comment]
Thank you. All fixed.

**Pull request merged**